media-tv/tvbrowser-bin/tvbrowser-bin-4.0.1.ebuild

```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2019 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit eutils desktop
+
+DESCRIPTION="Themeable and easy to use TV Guide - written in Java"
+HOMEPAGE="http://www.tvbrowser.org"
+MY_PN="tvbrowser"
+MY_P="${MY_PN}-${PV}"
+SRC_URI="mirror://sourceforge/project/tvbrowser/TV-Browser%20Releases%20%28Java%208%20and%20higher%29/${PV}/${MY_PN}_${PV}_bin.tar.gz"
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="virtual/jdk:1.8"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}"/"${MY_P}"
+
+generate_launcher() {
+	echo "#!/bin/bash"
+	echo "export JAVA_OPTS=\"\${JAVA_OPTS} -Dpropertiesfile=/opt/${P}/linux.properties\""
+	echo "/bin/bash /opt/${P}/tvbrowser.sh"
+}
+
+src_install()
+{
+	# Copy files and directories
+	cd "${S}"
+	dodir /opt/"${P}"
+	insinto /opt/"${P}"
+	doins -r *
+
+	# Generate launcher
+	local tvbrowser_bin="tvbrowser"
```

Comment:
In my opinion, what you're doing here is unneeded complexity, since it can be done a lot easier. 

https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-apps/baselayout-java/baselayout-java-0.1.0-r1.ebuild#n29 This is a nice example how to.
(@gyakovlev)

Reply:
This honestly does nothing, you can just use plain "tvbrowser" below instead of `"${tvbrowser_bin}"`.